Delete modal - create a shared delete modal, use for all deletions

himdel opened the pull request:
Fixes https://issues.redhat.com/browse/AAH-358

This unifies the confirmation modal for all deletions/removals in the app so that we're:

* using a confirmation modal,
* using patternfly title with `titleVariant='warning'` to fix the icon alignment in the header
* the title has the second word lowercased (Delete user => Delete group)
* there's a description, with the name of the deleted item in bold

It also adds a missing Delete button to the group detail screen (user detail has it),
tweaks `.gitignore` and `.prettierignore` to ignore vim swap files,
fixes a logo console warning,
and updates the Users tab in Group detail to work with 10+ users.
(I can split these off if needed :).)


**Before**:

Users > kebab Delete - icon alignment, missing bold

![usersb](https://user-images.githubusercontent.com/289743/109717579-24f4bc00-7b9e-11eb-8397-058cc89d7b88.png)

Users > detail > Delete - icon alignment, missing bold

![userdb](https://user-images.githubusercontent.com/289743/109717591-2a520680-7b9e-11eb-953e-43c9b03a09c8.png)

Groups > Delete - icon alignment, missing description

![groupsb](https://user-images.githubusercontent.com/289743/109717607-2faf5100-7b9e-11eb-9a59-3a3b4f2fefa3.png)

Groups > detail > Delete - no such button

![groupdetb](https://user-images.githubusercontent.com/289743/109717633-3938b900-7b9e-11eb-9aa9-51d25ff659f1.png)

Groups > detail > tab Users > kebab Remove - no confirmation modal, removed directly

![](https://i.imgur.com/DKUR9Tk.png)

---

**After**:

Users > kebab Delete

![users](https://user-images.githubusercontent.com/289743/109718001-b06e4d00-7b9e-11eb-8fd8-f9917c295e8f.png)

Users > detail > Delete

![userd](https://user-images.githubusercontent.com/289743/109718026-b6fcc480-7b9e-11eb-89ea-c1bd7b05863e.png)

Groups > Delete

![groups](https://user-images.githubusercontent.com/289743/109718050-bf54ff80-7b9e-11eb-9184-52dbc832476b.png)

(see comments for a version with a list/count of users)

Groups > detail > Delete

![groupdet](https://user-images.githubusercontent.com/289743/109718094-cbd95800-7b9e-11eb-9bd7-5270725aaa90.png)
![groupd](https://user-images.githubusercontent.com/289743/109718070-c67c0d80-7b9e-11eb-8970-597fafa4ec9f.png)

(see comments for a version with a list/count of users)

Groups > detail > tab Users > kebab Remove

![groupuser](https://user-images.githubusercontent.com/289743/109718121-d398fc80-7b9e-11eb-9e3b-904770e704ec.png)

---

~~Question: right now, a group can be deleted even if it has users assigned to it. Should that be possible? Warning at least?~~
Resolved by showing a list of affected users.

himdel commented:
@sbuenafe-rh can you review please? :) This should be unifying all the galaxy UI removal deletion confirmation modals and aligning them with patternfly guidelines.

ZitaNemeckova reviewed (approved):
LGTM 👍 I'd like to get Susan's approval before merge.

sbuenafe-rh commented:
@himdel The delete user and remove user from group modals look good. 

Deleting a group's message needs to be reworked a little to mention that the users in this group will lose access. Not sure if we can list out the users...

himdel commented:
@sbuenafe-rh thanks :) would something like this make sense?

for a bit, if the users list is not already loaded (the blue bit is a small spinner, spinning):

![spin](https://user-images.githubusercontent.com/289743/109996033-be3ee200-7d06-11eb-8f5c-be8b73dce4eb.png)

empty group:

![none](https://user-images.githubusercontent.com/289743/109996114-d282df00-7d06-11eb-9ee8-7aa617862dac.png)

nonempty group:

![some](https://user-images.githubusercontent.com/289743/109996133-d7e02980-7d06-11eb-9726-9c693c4a51e4.png)

Any ideas how to deal with groups with 10+ users?

(WIP: right now this is only implemented on the group detail screen, and not limiting the number of users to display in the list, nor the list height)

sbuenafe-rh commented:
@himdel looks good! For 10+ users, we could just state that deleting this group will affect [X] users.

himdel commented:
Thanks, updated... :)

For up to 10 items...

![10](https://user-images.githubusercontent.com/289743/110381586-94145980-8051-11eb-8a70-92eb65f54bdc.png)


Over 10...

![11](https://user-images.githubusercontent.com/289743/110381605-9bd3fe00-8051-11eb-8a92-487f3e021d26.png)



I've also fixed the Users tab in group detail to show more than 10 items, sorted by username,
moved the group deletion view logic to a shared modal and used this from the group-list, so both group deletions should behave the same now :).

Ready for more review :)

ZitaNemeckova commented on src/containers/group-management/delete-group-modal.tsx:
```diff
@@ -0,0 +1,51 @@
+  cancelAction: () => void;
+  deleteAction: () => void;
+  name: string;
+  users?: any[];
```
Use `UserType ` instead of `any`.

ZitaNemeckova commented on src/containers/group-management/group-list.tsx:
```diff
@@ -48,6 +47,8 @@ interface IState {
   groups: any[];
   createModalVisible: boolean;
+  deleteModalCount?: number;
+  deleteModalUsers?: any[];
```
This should be `UserType` as well.

ZitaNemeckova commented on src/loaders/standalone/standalone-loader.tsx:
```diff
@@ -163,15 +163,16 @@ class App extends React.Component<RouteComponentProps, IState> {
             })}
           >
-            <SmallLogo alt={APPLICATION_NAME}></SmallLogo>
+            {children}
```
🎉

ZitaNemeckova reviewed (approved):
LGTM :+1:

sbuenafe-rh commented:
@himdel When checking for affected users...will the Delete button be disabled?
